analysis/bach_open_taxonomy/bach_open_taxonomy/modelhub/aggregate.py

```diff
@@ -19,11 +22,31 @@ class Aggregate:
     def __init__(self, df):
         self._df = df
 
-    def _generic_aggregation(self, time_aggregation, column, filter, name):
-        if not time_aggregation:
-            time_aggregation = self._df._time_aggregation
-        gb = self._df.moment.dt.sql_format(time_aggregation) if time_aggregation else None
-        df = self._df.copy_override()
+    def _check_groupby(self,
+                       groupby,
```

Comment:
Let's define `GroupByType = Union[List[Union[str, Series]], str, Series, NotSet]` and add typing for the functions that use that param.
```suggestion
             groupby: GroupByType,
```